docs/adr/0003-event-broker-for-subscription-platform.md

```diff
@@ -0,0 +1,108 @@
+# Event Broker for Subscription Platform
+
+* Deciders: Ben Bangert, Phil Booth, Ryan Kelly, Shane Tomlinson
+* Date: 2019-05-14
+
+## Context and Problem Statement
+
+External OAuth Relying Parties (RPs) that Firefox Account users authenticate
+to need to be kept up to date about whether the user still has an active
+subscription as well as knowing when to delete the user data. For internal
+Mozilla use, Mozilla provides [Firefox Service Notifications][] over SQS
+queues. These events are intended for internal trusted use and not suitable
+for external RPs.
+
+A solution for these concerns is a new system referred to as the FxA Event
+Broker which will store FxA related events and distribute them to relevant
+RPs that the user has accessed via webhooks. The initial version described here applies primarily to external RPs which will only receive subscription status
+changes and account deletion events. This solution is being built with future
+expansion in mind which will require event stream storage per user and
+notification of internal Mozilla RPs.
+
+Difficulties inherent in this solution lie in where to store the source of
+truth for what RPs a user has authenticated, where the delivery functionality
+should reside, where the event streams for a user should reside, what set of
+data should be included in [Firefox Service Notifications][], and how this
+data should be communicated to the delivery system to avoid additional API
+query load on existing FxA services.
+
+## Decision Drivers
+
+* Subscription services deadlines
+* Effort required and experience available for FxA changes
+* Separation of concerns from existing FxA microservices
+* Difficulty of schema migrations in existing FxA microservices
+* Suitability of existing FxA databases for large-scale event storage
```

Comment:
Is it worth saying something here about the architectural desire to treat "auth" and "oauth" as a single service over time?

Reply:
Yep, I added a comment.